### PR TITLE
Update 05.dl-complexities.md

### PR DESCRIPTION
**Pull request by mdkessler**
**Did you add yourself as a [contributor](https://github.com/Benjamin-Lee/deep-rules/blob/master/contributors.md) if this is your first contribution?**

- [x] Yes, I added myself or am already a contributor

**Any more details?**

### CODE REVIEW

**Comment by Benjamin-Lee** [comment]
FYI, for reviewers, take a look at the rich diff, not the source diff. I'm not sure why the source diff is such a mess to look at.

File: content/05.dl-complexities.md [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
+This can help to prevent practitioners from wasting signficant time and resources on running highly complex models that feature numerous unresolved problems. 
+Again, beware of the choices that are being made implicitly (_i.e._ by default settings) by your framework of choice (e.g. choice of optimization algorithm), as these seemingly trivial specifics can actually have signficant effects on model performance.
+For example, adaptive methods often lead to faster convergence during training, but may lead to worse generalization performance on independent datasets [@url:https://papers.nips.cc/paper/7003-the-marginal-value-of-adaptive-gradient-methods-in-machine-learning]).
+These nuanced elements are easy to overlook, but it is critical to carefully consider them and to evaluate their potential impact (see [Tip 6](#hyperparameters)).
 
-In short, use smaller and simpler networks to enable faster prototyping and follow general software development best practices to maximize reproducibility.
+In short, use smaller and simpler networks to enable faster prototyping, and follow general software development best practices to maximize reproducibility.
```
Comment (Benjamin-Lee) [inline content/05.dl-complexities.md:20]: I think we do need the comma here since both verbs refer to the same subject:

```suggestion
In short, use smaller and simpler networks to enable faster prototyping and follow general software development best practices to maximize reproducibility.
```

File: content/05.dl-complexities.md [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
 
-Similar to [Tip 4](#baselines), try to start with a relatively small network and increase the size and complexity as needed to prevent wasting time and resources. 
-Beware of the seemingly trivial choices that are being made implicitly by default settings in your framework of choice e.g. choice of optimization algorithm (adaptive methods often lead to faster convergence during training but may lead to worse generalization performance on independent datasets [@url:https://papers.nips.cc/paper/7003-the-marginal-value-of-adaptive-gradient-methods-in-machine-learning]).
-These need to be carefully considered and their impacts evaluated (see [Tip 6](#hyperparameters)).
+Similar to the suggestions in [Tip 2](#baselines) about starting with simpler models, try to start with a relatively small network, and to then increase the size and complexity as needed.
```
Comment (Benjamin-Lee) [inline content/05.dl-complexities.md:14]: This will be changed when we standardize on third person but it should probably skip the second "to" for parallel structure and remove the comma since the two verbs refer to the same subject (the implied "you"):

```suggestion
Similar to the suggestions in [Tip 2](#baselines) about starting with simpler models, try to start with a relatively small network and then increase the size and complexity as needed.
```

**Review by Benjamin-Lee — CHANGES_REQUESTED** [review]
Some minor grammatical suggestions but overall it improves the flow of the tip. Good work!

File: content/05.dl-complexities.md [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
-All code, random seeds, parameters, and results must be carefully corralled using general good coding practices (for example, version control [@doi:10.1371/journal.pcbi.1004947], continuous integration etc.) in order to be effective and interpretable.
-This organization is also key to being able to efficiently share and reproduce your work [@doi:10.1371/journal.pcbi.1003285; @arxiv:1810.08055] as well as to update your model as new data becomes available.
+Correctly training deep neural networks is a non-trivial process, as there are many different options and potential pitfalls at every stage.
+To get good results, you must expect to train many networks across a range of different parameter and hyperparameter settings.
+Such training can be made more difficult by the demanding nature of these deep networks, which often require extensive computing infrastructure and optimization in order to achieve state-of-the-art performance [@doi:10.1109/JPROC.2017.2761740].
+Furthermore, this experimentation is often noisy, which necessitates increased repitition, and furthers the organizational challenges inherent to DL.
```
Comment (mdkessler) [inline content/05.dl-complexities.md:6]: Good catch, further is repeated. I tried to avoid increases since "increased" is 3 words before. How about "exacerbates", as below?

```suggestion
Furthermore, this experimentation is often noisy, which necessitates increased repitition, and exacerbates the organizational challenges inherent to DL.
```

**Review by Benjamin-Lee — APPROVED** [review]

**Review by betsig — APPROVED** [review]
